app/util/PendingAction.scala

```diff
@@ -10,11 +10,11 @@ trait PendingAction[R] {
   /**
     * Completes the action.
     */
-  def complete(): Try[R]
+  def complete(implicit ec: ExecutionContext): Future[R]
 
   /**
     * Cancels the action.
     */
-  def cancel()
+  def cancel(implicit ec: ExecutionContext)
```

Comment:
This is a side-effecting method, so reintroducing a the standard parenthesis would be nice.

Reply:
I'd also remove the execution context from trait methods, I'd rather inject it into the implementations. Also: public methods should have explicit return types.